gobblin-data-management/src/main/java/gobblin/data/management/copy/replication/CopySourceWithMultipleSourceLocation.java

```diff
@@ -0,0 +1,276 @@
+    //      else{
+    //        throw new RuntimeException("Can not find config store root");
+    //      }
+    //
+    //      //public Collection<URI> getImportedBy(URI configKeyUri, boolean recursive)
+    //      Collection<URI> importedBy = configClient.getImportedBy(new URI(storeRoot+"/tag/testTag"), true);
+    //      for(URI u: importedBy){
+    //        log.info("AAA imported by is " + u);
+    //      }
+    //
+    //      log.info("generate workunit for /data/derived/data_derived_test_src");
+    //      Config c = configClient.getConfig(storeRoot+"/data/derived/data_derived_test_src/");
+    //      //displayAllConfig(c);
+    //      ReplicationConfiguration rc = ReplicationConfiguration.buildFromConfig(c);
+    //
+    //      generateWorkunitsPerReplicationDataset(originState,workUnitList, minWorkUnitWeight,  rc);
+    //
+    //      log.info("generate workunit for /data/derived/supertitleToSkills");
+    //      c = configClient.getConfig(storeRoot+"/data/derived/supertitleToSkills/");
+    //      rc = ReplicationConfiguration.buildFromConfig(c);
+    //      return generateWorkunitsPerReplicationDataset(originState,workUnitList, minWorkUnitWeight,  rc);
+    //
+    //
+    //    } catch (VersionDoesNotExistException | ConfigStoreFactoryDoesNotExistsException | ConfigStoreCreationException
+    //        | URISyntaxException e) {
+    //      log.error("Caught error " + e.getMessage());
+    //      throw new RuntimeException(e);
+    //    }
+  }
+
+  private List<Config> generateReplicationConfigs(final SourceState originState) {
+    try {
+      String storeRoot;
+      String replicationRootPath;
+      String replicationTag;
+      ConfigClient configClient = ConfigClient.createConfigClient(VersionStabilityPolicy.WEAK_LOCAL_STABILITY);
+      if (originState.contains(CONFIG_STORE_ROOT) && originState.contains(CONFIG_STORE_REPLICATION_ROOT)
+          && originState.contains(CONFIG_STORE_REPLICATION_TAG)) {
+        storeRoot = originState.getProp(CONFIG_STORE_ROOT);
+        replicationRootPath = new URI(storeRoot + originState.getProp(CONFIG_STORE_REPLICATION_ROOT)).getPath();
+        replicationTag = storeRoot + originState.getProp(CONFIG_STORE_REPLICATION_TAG);
+      } else {
+        throw new RuntimeException(String.format("Can not find required config values for %s, %s, %s",
+            CONFIG_STORE_ROOT, CONFIG_STORE_REPLICATION_ROOT, CONFIG_STORE_REPLICATION_TAG));
+      }
+
+      // get all the URIs which imports {@link #replicationTag}
+      Collection<URI> importedBy = configClient.getImportedBy(new URI(replicationTag), true);
+
+      List<URI> replicationDataset = new ArrayList<URI>(importedBy);
+      Comparator<URI> pathLengthComparator = new Comparator<URI>() {
+        public int compare(URI c1, URI c2) {
+          return c1.getPath().length() - c2.getPath().length();
+        }
+      };
+
+      // sort the URI based on the path length to make sure the parent path appear before children
```

Comment:
It's hard to review this part since I am not completely sure what CONFIG_STORE_REPLICATION_ROOT is .